apps/src/lib/kits/maker/CircuitPlaygroundBoard.js

```diff
@@ -0,0 +1,193 @@
+/** @file Board controller for Adafruit Circuit Playground */
+import {EventEmitter} from 'events'; // provided by webpack's node-libs-browser
+import ChromeSerialPort from 'chrome-serialport';
+import five from 'johnny-five';
+import Playground from 'playground-io';
+import {
+  initializeCircuitPlaygroundComponents,
+  componentConstructors
+} from './PlaygroundComponents';
+import {BOARD_EVENT_ALIASES} from './PlaygroundConstants';
+
+// Polyfill node's process.hrtime for the browser, gets used by johnny-five.
+process.hrtime = require('browser-process-hrtime');
+
+/** @const {number} serial port transfer rate */
+const SERIAL_BAUD = 57600;
+
+const J5_CONSTANTS = {
+  INPUT: 0,
+  OUTPUT: 1,
+  ANALOG: 2,
+  PWM: 3,
+  SERVO: 4
+};
+
+/**
+ * Controller interface for an Adafruit Circuit Playground board using
+ * Circuit Playground Firmata firmware.
+ */
+export default class CircuitPlaygroundBoard extends EventEmitter {
+  constructor(portName) {
+    super();
+
+    /** @private {string} a port identifier, e.g. "/dev/ttyACM0" */
+    this.portName_ = portName;
+
+    /** @private {five.Board} A johnny-five board controller */
+    this.fiveBoard_ = null;
+
+    /** @private {Object} Map of component controllers */
+    this.prewiredComponents_ = null;
+  }
+
+  /**
+   * Open a connection to the board on its configured port.
+   * @return {Promise}
+   */
+  connect() {
+    return Promise.resolve()
+        .then(() => this.connectToFirmware())
+        .then(() => this.initializeComponents())
+        .then(() => this.initializeEventForwarding());
+  }
+
+  /**
+   * Open the serial port connection and create a johnny-five board controller.
+   * Exposed as a separate step here for the sake of the setup page; generally
+   * recommended to just call connect(), above.
+   * @return {Promise}
+   */
+  connectToFirmware() {
```

Comment:
So basically `connectToFirmware` becomes 'auto-bound'?  I'm not sure I love this syntax, but I'm willing to try it.